### PR TITLE
adding requires field to output_spec

### PR DESCRIPTION
**Pull request by djarecka**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

- adding requires field to `output_spec`,
- simplifying functions in helpers (removing ` output_names_from_inputfields`)

## Checklist

- [ ] All tests passing
- [ ] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [ ] My code follows the code style of this project
(we are using `black`: you can `pip install pre-commit`,
run `pre-commit install` in the `pydra` directory
and `black` will be run automatically with each commit)

## Acknowledgment
- [ ] I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: pydra/engine/specs.py [hunk 1]
```diff
@@ -449,6 +449,10 @@ def _field_defaultvalue(self, fld, output_dir):
 
     def _field_metadata(self, fld, inputs, output_dir):
         """Collect output file if metadata specified."""
+        if "requires" in fld.metadata:
+            for inp in fld.metadata["requires"]:
+                if getattr(inputs, inp) in [attr.NOTHING, None, False]:
```
Comment (satra) [inline pydra/engine/specs.py:454]: `False` could be a valid state for a field. 

```suggestion
                if getattr(inputs, inp) in [attr.NOTHING, None]:
```

File: pydra/engine/specs.py [hunk 2]
```diff
@@ -461,7 +465,7 @@ def _field_metadata(self, fld, inputs, output_dir):
         elif "callable" in fld.metadata:
             return fld.metadata["callable"](fld.name, output_dir)
```
Comment (satra) [inline pydra/engine/specs.py:466]: do we have this documented somewhere?

Comment (djarecka) [reply pydra/engine/specs.py:466]: only tests probably...

File: pydra/engine/specs.py (hunk 1 above)
Comment (djarecka) [reply pydra/engine/specs.py:454]: but usually you use flag that are bool

**Comment by djarecka** [comment]
will open a new one

**Pull request closed without merging**